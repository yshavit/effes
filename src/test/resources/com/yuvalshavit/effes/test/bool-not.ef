def not (bool: True | False) -> (True | False):
  return case bool of
    True: False
    False: True

def main -> Void:
  print not True
  print not False
  return Void
