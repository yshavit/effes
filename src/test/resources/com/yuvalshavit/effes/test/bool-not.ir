def not (bool: False | True) -> False | True:
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: bool
          pos: 0
      of True:
        CtorInvoke
          type: False
      of False:
        CtorInvoke
          type: True
