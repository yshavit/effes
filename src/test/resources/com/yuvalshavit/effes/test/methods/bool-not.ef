not (bool: True | False) -> (True | False):
  return case bool of
    True: False
    False: True

main -> Void:
  debugPrint not True
  debugPrint not False
  return Void
