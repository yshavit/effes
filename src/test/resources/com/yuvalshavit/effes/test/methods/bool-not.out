Your argument was: False
Your argument was: True
>> Void
